core/aws.py

```diff
@@ -27,38 +27,34 @@
     )
 
 
-def upload_file(file: IO, bucket: str, object_name: str) -> bool:
+def upload_file(file: IO, bucket: str, object_name: str, metadata: dict = None) -> bool:
     """Uploads a file to an S3 bucket.
 
     :param file: a readable file-like object
     :param bucket: bucket to upload to
     :param object_name: S3 object name
+    :param metadata: optional dictionary containing metadata for upload
     :return: True if successful else False
     """
     file.seek(0)
-    try:
-        _S3_CLIENT.upload_fileobj(file, bucket, object_name)
-    except (ClientError, EndpointConnectionError) as bucket_error:
-        current_app.logger.error(bucket_error)
-        return False
+    _S3_CLIENT.upload_fileobj(file, bucket, object_name, ExtraArgs={"Metadata": metadata if metadata else {}})
     return True
 
 
-def get_file(bucket: str, object_name: str) -> BytesIO | None:
+def get_file(bucket: str, object_name: str) -> tuple[BytesIO, dict] | None:
     """Retrieves a file from an S3 bucket.
 
     :param bucket: bucket to retrieve from
     :param object_name: S3 object name
     :return: retrieved file as a BytesIO
     """
-    file = BytesIO()
     try:
-        _S3_CLIENT.download_fileobj(bucket, object_name, file)
+        response = _S3_CLIENT.get_object(Bucket=bucket, Key=object_name)
     except (ClientError, EndpointConnectionError) as bucket_error:
         current_app.logger.error(bucket_error)
         return None
```

Comment:
Would it be better to re-throw the exception after logging. Is it ambiguous to return `None`?